fastmri_recon/data/datasets/multicoil/preprocessing.py

```diff
@@ -33,6 +36,58 @@ def from_kspace_to_masked_kspace_and_mask(images, kspaces):
             return (kspaces_channeled, mask, smaps), images_channeled
     return from_kspace_to_masked_kspace_and_mask
 
+
+def non_cartesian_from_kspace_to_nc_kspace_and_traj(
+        nfft_ob,
+        image_size,
+        acq_type='radial',
+        scale_factor=1,
+        compute_dcomp=False,
+        parallel=False,
+        **acq_kwargs
+):
+    def from_kspace_to_nc_kspace_and_traj(images, kspaces):
+        if acq_type == 'radial':
+            traj = get_radial_trajectory(image_size, **acq_kwargs)
+        elif acq_type == 'cartesian':
+            traj = get_debugging_cartesian_trajectory()
+        else:
+            raise NotImplementedError(f'{acq_type} dataset not implemented yet.')
+        interpob = nfft_ob._extract_nufft_interpob()
+        nufftob_forw = kbnufft_forward(interpob)
+        nufftob_back = kbnufft_adjoint(interpob)
+        dcomp = calculate_radial_dcomp_tf(
```

Comment:
do this only if `compute_dcomp`

Reply:
Ahh yes, I was searching for its parameter.
However, this might modify the code a bit more, would you estimate smaps with dens_comp set to 1 in case you wont use dens_comp? That isnt a good idea in my opinion, the Smaps in that case will not be good..

Reply:
I see your point. Maybe then we could just have density compensation all the time when in multichannel since we already make the point that they are useful in the single coil case.

Reply:
Made changes to dataset codes to fix this

Reply:
Hmm then the compute dcomp arg shouldn't be present at all

Reply:
I agree to that. However, one way you could work is to instance it to be an identity matrix in case you it is disabled. 
That way, while Smaps wont be great (in 3D this could be really bad) , but 2D it shall be fine.

Reply:
yup yup but I'd rather not have it around.